### ISSUE TITLE
onnx/onnx/backend/test/case/node/gather.py has different behavior on Windows and Linux

### ISSUE THREAD

**Issue by snnn** [onnx/onnx#813]
indices is 32 bits on windows but 64 bits on Linux

https://stackoverflow.com/questions/36278590/numpy-array-dtype-is-coming-as-int32-by-default-in-a-windows-10-64-bit-machine

**Comment by houseroad** [comment]
Yes, this code is not clear enough. Made it clear in https://github.com/onnx/onnx/pull/815
